src/target/riscv/riscv-013.c

```diff
@@ -977,27 +977,51 @@ static int execute_abstract_command(struct target *target, uint32_t command,
 	return ERROR_OK;
 }
 
-static riscv_reg_t read_abstract_arg(struct target *target, unsigned index,
-		unsigned size_bits)
+static void add_data_regs_read(struct riscv_batch *batch, unsigned int index,
+		unsigned int size_bits)
 {
+	const unsigned int size_in_words = size_bits / 32;
+	assert(size_bits % 32 == 0);
```

Comment:
I believe you can add one more assert here:

```suggestion
	assert(size_bits % 32 == 0);
	assert(size_bits >= 32);	
```

Reply:
Also, I believe the functions are bit more readable when the assertions on input parameters are the very first statements of the function body.

If you are in favor of that, you can make this change here and also in the other functions.

Reply:
> I believe you can add one more assert here:

Addressed.